### PR TITLE
Fix caching bug for suspended users

### PR DESCRIPTION
**Pull request by hemant10yadav**
https://dimagi.atlassian.net/browse/CCCT-405

### CODE REVIEW

**Comment by pxwxnvermx** [comment]
It would be better to remove the caching here and making the `get_opp_access` a method on the `OpportunitySerializer` class with the django [cached_property](https://docs.djangoproject.com/en/5.0/ref/utils/#django.utils.functional.cached_property) decorator to avoid fetching the opportunity access multiple times. 

The caching here was intially added to accomplish the same goal of avoiding multiple fetches of opportunity access for getting the various properties required on the OpportunitySerializer which can be done better by using the `cached_property` decorator.

@hemant10yadav

**Review by pxwxnvermx — APPROVED** [review]

**Review by calellowitz — APPROVED** [review]

File: commcare_connect/opportunity/views.py [hunk 1]
```diff
@@ -889,6 +890,10 @@ def suspend_user(request, org_slug=None, opp_id=None, pk=None):
     access.suspension_date = now()
     access.suspension_reason = request.POST.get("reason", "")
     access.save()
+
+    # Clear the cached opportunity access for the suspended user
+    remove_opportunity_access_cache(access.user, access.opportunity)
```
Comment (sravfeyn) [inline commcare_connect/opportunity/views.py:895]: You can call `_get_opp_access.clear` directly from within `access.save`

Comment (hemant10yadav) [reply commcare_connect/opportunity/views.py]: @sravfeyn  Changed

**Comment by calellowitz** [comment]
Let's offline about that last commit. I prefer it the way it was before, because we only want to invalidate it at the moment the user is suspended, not every time we save if the user is suspended. This does make me realize we should also do it when they are unsuspended as well, but I would prefer to have that in the relevant method as well, rather than in the save function.

**Review by calellowitz — APPROVED** [review]

**Pull request merged**